Added option to overwrite existing stored credentials.

lraccomando opened the pull request:
Added an option to django_orm.Storage locked_put method to allow for overwriting an existing credential record rather than the creation of a new one.

I needed this functionality on a project I'm working on and ended up subclassing Storage in order to achieve it. Thought it may be useful for others as well. The default behavior remains the same.

lraccomando committed:
Added option to overwrite existing stored credentials.

Added an option to django_orm.Storage locked_put method to allow for
overwriting an existing credential record rather than the creation of a
new one.

craigcitro commented on oauth2client/django_orm.py:
```diff
@@ -116,14 +116,19 @@ def locked_get(self):
     """Write a Credentials to the datastore.
 
     Args:
       credentials: Credentials, the credentials to store.
```
Could you add a description of the new argument to the docstring?

craigcitro commented on oauth2client/django_orm.py:
```diff
@@ -116,14 +116,19 @@ def locked_get(self):
-    entity = self.model_class(**args)
+    
+    if overwrite: 
+      entity, is_new = self.model_class.objects.get_or_create(**args)
```
Did you want to use `is_new`? If not, maybe just call it `unused_is_new` or `_`?

lraccomando committed:
Wrote doc string for new storage parameter.

Added proper doctoring for the overwrite parameter in Django ORM
storage and updated an unused variable.

lraccomando committed:
Merge branch 'master' of git://github.com/google/oauth2client
